extension/content/components/recipes/form/arguments/ShowHeartBeat.js

```diff
@@ -61,21 +61,35 @@ function RepeatOptionsField() {
     },
   ];
 
-  const repeatsEveryChangeSideEffect = ({ data, value }) => {
+  const repeatOptionChangeSideEffect = ({ data, value }) => {
     if (value !== "xdays") {
       /* eslint-disable no-unused-vars */
       const {
-        repeatEvery: _omitRepeatOption,
+        repeatEvery: _omitRepeatEvery,
         action,
         ...cleanedArguments
       } = data.arguments;
       /* eslint-enable no-unused-vars */
       return { ...data, arguments: cleanedArguments };
     }
 
-    return data;
+    return { ...data, arguments: { ...data.arguments, repeatEvery: 1 } };
   };
 
+  return (
+    <>
+      <SelectField
+        changeSideEffect={repeatOptionChangeSideEffect}
+        label="How often should the prompt be shown?"
+        name="repeatOption"
+        options={SHOW_PROMPT_OPTIONS}
+      />
+      {repeatEveryOption()}
+    </>
+  );
+}
+
+function repeatEveryOption() {
```

Comment:
almost there. this should be a functional component so something like `RepeatEveryField` not `repeatEveryOption`.